src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -672,8 +674,24 @@ public function load(array $configs, ContainerBuilder $container): void
                 }
                 $tagAttributes['method'] = $reflector->getName();
                 $tagAttributes['class'] = $reflector->getDeclaringClass()->name;
+                $reflector = $reflector->getDeclaringClass();
             }
-            $definition->addTag('container.factory', $tagAttributes);
+            if (null === $tagAttributes['class'] && null === $tagAttributes['service'] && null === $tagAttributes['expression']) {
+                throw new LogicException(sprintf('Factory attribute must declare a class, service or expression in %s.', $reflector->name));
+            }
+            if (null === $tagAttributes['method'] && (null !== $tagAttributes['class'] || null !== $tagAttributes['service'])) {
+                $tagAttributes['method'] = '__invoke';
+            }
+            // Prevent using both #[Factory] and #[Autoconfigure(constructor:...)]
+            $autoconfigureWithConstructor = array_filter(
+                $reflector->getAttributes(Autoconfigure::class, \ReflectionAttribute::IS_INSTANCEOF),
+                static fn(\ReflectionAttribute $attr) => null !== $attr->newInstance()->constructor
+            );
+            if (0 < \count($autoconfigureWithConstructor)) {
+                throw new AutoconfigureFailedException($reflector->name, sprintf('Using both attributes #[Factory] and #[Autoconfigure(constructor: ...)] on is not allowed in %s.', $reflector->name));
+            }
+            $tagAttributes['arguments'] = array_map(fn ($argument): string => $argument instanceof Reference ? ('@'.$argument) : $argument, $tagAttributes['arguments']);
```

Comment:
so you handle only one specific case ? when doing `new Reference` you can pass a second argument `invalid behaviour`. Also what about parameters ?

Reply:
You're right I missed a lot of cases. While adding some obvious ones (Parameter, Expression and Reference) I've realized handling all the possible arguments is a lot of work.  So I went back to the good old 'Yaml hack' : 
- Using the `YamlDumper` to convert the arguments to valid yaml strings in the `registerAttributeForAutoconfiguration` callback.
-  Using `Yaml\Parser` in the compiler pass to convert them back to `TaggedValue` when needed.

I think going with tags is complicating the process a lot. I thought about what you suggested earlier : store the `#[Factory]` attribute properties in a dedicated service to fetch them later. But that means the tag would not be usable on its own.  
So maybe we should just consider this feature is "attribute-only" and stop using the tag, simply fetch the data from the attribute in the compiler pass and use `setFactory` and `setArguments` directly.

Reply:
I wouldn't mind this being an attribute only syntax. But not sure how the symfony team will see this. Why I am confortable with this is : it is already easy to define in yaml / xml / php files but there is no simple way for attributes. to each their own logic is not a problem IMO.